src/math/p5.Vector.js

```diff
@@ -326,6 +326,138 @@ p5.Vector.prototype.add = function add(x, y, z) {
   return this;
 };
 
+/// HELPERS FOR REMAINDER METHOD
+const calculateRemainder2D = function(xComponent, yComponent) {
+  if (xComponent === 0 && yComponent === 0) {
+    return this;
+  } else if (xComponent === 0) {
+    this.y = this.y % x[1];
+    return this;
+  } else if (yComponent === 0) {
+    this.x = this.x % x[0];
+    return this;
+  } else {
+    this.x = this.x % xComponent;
+    this.y = this.y % yComponent;
+    return this;
+  }
+};
+
+const calculateRemainder3D = function(xComponent, yComponent, zComponent) {
+  if (xComponent === 0 && yComponent === 0 && zComponent === 0) {
+    return this;
+  } else if (xComponent === 0 && yComponent !== 0 && zComponent !== 0) {
+    this.y = this.y % yComponent;
+    this.z = this.z % zComponent;
+    return this;
+  } else if (xComponent !== 0 && yComponent === 0 && zComponent !== 0) {
+    this.x = this.x % xComponent;
+    this.z = this.z % zComponent;
+    return this;
+  } else if (xComponent !== 0 && yComponent !== 0 && zComponent === 0) {
+    this.x = this.x % xComponent;
+    this.y = this.y % yComponent;
+    return this;
+  } else if (xComponent === 0 && yComponent === 0 && zComponent !== 0) {
+    this.z = this.z % zComponent;
+    return this;
+  } else if (xComponent === 0 && yComponent !== 0 && zComponent === 0) {
+    this.y = this.y % yComponent;
+    return this;
+  } else if (xComponent !== 0 && yComponent === 0 && zComponent === 0) {
+    this.x = this.x % xComponent;
+    return this;
+  } else {
+    this.x = this.x % xComponent;
+    this.y = this.y % yComponent;
+    this.z = this.z % zComponent;
+    return this;
```

Comment:
could this whole block be rewritten more simply as:
```js
if (xComponent !== 0 ) {
  this.x = this.x % xComponent;
}
if (yComponent !== 0 ) {
  this.y = this.y % yComponent;
}
if (zComponent !== 0 ) {
  this.z = this.z % zComponent;
}
return this;
```

Reply:
and same with 2D method above?

Reply:
made necessary changes bf25493